binary_search_tree/tree.py

```diff
@@ -14,44 +14,147 @@ class Tree:
+        print(traversal_list)
+        return traversal_list
 
-    # Time Complexity: 
-    # Space Complexity:     
+    def preorder_helper(self, current, traversal_list):
+        if current:
+            traversal_list.append(
+                {
+                    "key": current.key,
+                    "value": current.value
+                }
+            )
+            self.preorder_helper(current.left, traversal_list)
+            self.preorder_helper(current.right, traversal_list)
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)    
     def preorder(self):
-        pass
+        traversal_list = []
+        self.preorder_helper(self.root, traversal_list)
+        return traversal_list
+
+    def postorder_helper(self, current, traversal_list):
+        if current:
+            self.postorder_helper(current.left, traversal_list)
+            self.postorder_helper(current.right, traversal_list)
+            traversal_list.append(
+                {
+                    "key": current.key,
+                    "value": current.value
+                }
+            )
 
-    # Time Complexity: 
-    # Space Complexity:     
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def postorder(self):
-        pass
+        traversal_list = []
+        self.postorder_helper(self.root, traversal_list)
+        return traversal_list
 
-    # Time Complexity: 
-    # Space Complexity:     
+    def height_helper(self, current, height_list, height):
+        if current:
+            height += 1
+            self.height_helper(current.left, height_list, height)
+            self.height_helper(current.right, height_list, height)
+        height_list.append(height)
+
+    # Not sure this is the best way to find height
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def height(self):
-        pass
+        height_list = []
+        height = 0
+        self.height_helper(self.root, height_list, height)
+        return max(height_list)
 
 
-#   # Optional Method
-#   # Time Complexity: 
-#   # Space Complexity: 
-    def bfs(self):
-        pass
+    def bfs_helper(self, current, traversal_list):
+        if current:
+            if current.left:
+                traversal_list.append(
+                {
+                    "key": current.left.key,
+                    "value": current.left.value
+                }
+            )
+            if current.right:
+                traversal_list.append(
+                {
+                    "key": current.right.key,
+                    "value": current.right.value
+                }
+            )
+            self.bfs_helper(current.left, traversal_list)
+            self.bfs_helper(current.right, traversal_list)
 
-        
+    # This passes the tests, but I'm not sure if it would actually work with a more complex tree...
+    # Optional Method
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
+    def bfs(self):
```

Comment:
You're right I don't think this will quite work as advertised on a more complicated tree.  I think you would need to change lines 140-141 to use a queue to determine which node to call the bfs helper on.  

I would suggest a 2nd list, as a queue, and you remove the front item from the queue to determine which node to call bfs on next.  Further on  the above if statements you would add `current.left` and `current.right` to the back of the queue.